src/database/UIDatabase.js

```diff
@@ -35,36 +33,53 @@ export class UIDatabase {
     this.database = database;
   }
 
+  ERRORS = {
+    ERROR_IN_TRANSACTION: { code: 'ERROR_IN_TRANSACTION', message: 'Database is in a transaction' },
+    ERROR_NO_PERMISSION: { code: 'ERROR_NO_PERMISSION', message: 'Storage permission not granted' },
+    ERROR_UNKNOWN: { code: 'ERROR_UNKOWN', message: 'ERROR_UNKNOWN' },
+  };
+
   /**
-   * Closes the database, exports the realm file to '/Download/mSupplyMobile\ data' on device
-   * file system. The app will crash if anything tries to access the database while it is closed.
+   * Exports the realm file to '/Download/mSupplyMobile\ data' on device file system.
+   * Ensures there is enough space, the realm exists and requests storage permission,
+   * if required.
    */
-  exportData(filename = 'msupply-mobile-data') {
-    // TO DO: add method to react-native-database.
 
+  async exportData(filename = 'msupply-mobile-data') {
     const { realm } = this.database;
-    const realmPath = realm.path;
+    const { path: realmPath } = realm;
     const exportFolder = `${RNFS.ExternalStorageDirectoryPath}/Download/mSupplyMobile_data`;
+    const copyFileName = `${filename}${formatDate(new Date(), 'dashes')}`.replace(
+      /[~\\\\/|?*<:>"+]/g,
```

Comment:
I think in general good to give regexes comments.

Reply:
definitely, cheers 842c72e1047ef6f3a686a43ee9ee7bdf9763e37b